pandas/tests/io/test_sql.py

```diff
@@ -1813,6 +1814,25 @@ def main(connectable):
         DataFrame({"test_foo_data": [0, 1, 2]}).to_sql("test_foo_data", self.conn)
         main(self.conn)
 
+    @pytest.mark.parametrize(
+        "input",
+        [{"foo": [np.inf]}, {"foo": [-np.inf]}, {"foo": [-np.inf], "infe0": ["bar"]}],
+    )
+    def test_to_sql_with_negative_npinf(self, input):
+        # GH 34431
+
+        df = pd.DataFrame(input)
+        df.to_sql("foobar", self.conn, index=False)
+        try:
+            res = sql.read_sql_table("foobar", self.conn)
+            tm.assert_equal(df, res)
+        except ValueError as err:
```

Comment:
Instead just check for which DB is being tested and for mysql use `pytest.raises`